cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/ReadSegmentRangeCommand.java

```diff
@@ -42,29 +47,97 @@ public ReadSegmentRangeCommand(CommandArgs args) {
+        ensureArgCount(5);
 
         final String fullyQualifiedSegmentName = getArg(0);
         final int offset = getIntArg(1);
         final int length = getIntArg(2);
         final String segmentStoreHost = getArg(3);
+        final String fileName = getArg(4);
+
+        Preconditions.checkArgument(offset >= 0, "The provided offset cannot be negative.");
+        Preconditions.checkArgument(length >= 0, "The provided length cannot be negative.");
+
         @Cleanup
         CuratorFramework zkClient = createZKClient();
         @Cleanup
         SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient);
-        CompletableFuture<WireCommands.SegmentRead> reply = segmentHelper.readSegment(fullyQualifiedSegmentName,
-                offset, length, new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), "");
-        WireCommands.SegmentRead segmentRead = reply.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
-        output("ReadSegment: %s", segmentRead.toString());
-        output("SegmentRead content: %s", segmentRead.getData().toString(StandardCharsets.UTF_8));
+        readAndWriteSegmentToFile(segmentHelper, segmentStoreHost, fullyQualifiedSegmentName, offset, length, fileName);
+        output("\nThe segment data has been successfully written into %s", fileName);
+    }
+
+    /**
+     * Creates the file (and parent directory if required) into which thee segment data is written.
+     *
+     * @param fileName
+     * @return A {@link File} object representing the filename provided.
+     * @throws FileAlreadyExistsException if the file already exists, to avoid any accidental overwrites.
+     * @throws IOException if the file/directory creation fails.
+     */
+    private File createFileAndDirectory(String fileName) throws IOException {
+        File f = new File(fileName);
+        // If file exists throw FileAlreadyExistsException, an existing file should not be overwritten with new data.
+        if (f.exists()) {
+            throw new FileAlreadyExistsException("Cannot write segment data into a file that already exists.");
+        }
+        if (!f.getParentFile().exists()) {
+            f.getParentFile().mkdirs();
+        }
+        f.createNewFile();
+        return f;
+    }
+
+    /**
+     * Reads the contents of the segment starting from the given offset and writes into the provided file.
+     *
+     * @param segmentHelper             A {@link SegmentHelper} instance to read the segment.
+     * @param segmentStoreHost          Address of the segment-store to read from.
+     * @param fullyQualifiedSegmentName The name of the segment.
+     * @param offset                    The starting point from where the segment is to be read.
+     * @param length                    The number of bytes to read.
+     * @param fileName                  A name of the file to which the data will be written.
+     * @throws IOException if the file create/write fails.
```

Comment:
If you have this many exceptions, I may recommend you just say `throws Exception` and then in your Javadoc you only talk about that one.  There is no useful information that someone might get from this Javadoc (as currently written) if all the exceptions have "if the request fails". The point of separating exceptions out is for the caller to write handling code (i.e., `try-catch`) specifically for those exceptions and figure out what to do in each case.

Reply:
Shortened all the exceptions in regards to the failure of the request into a single `throws Exception`. This leaves the javadoc going into two possible exceptions here:

1. `IOException` when anything happens in regards to file creation and handling.
2. `Exception` for any other failures that can be thrown while making the request.